packages/x-date-pickers/src/internals/components/PickerProvider.tsx

```diff
@@ -54,6 +54,11 @@ export interface PickerContextValue {
    * @param {React.UIEvent} event The DOM event that triggered the change.
    */
   onClose: (event: React.UIEvent) => void;
+  /**
+   * Close the picker if it's open, open it if it's closed.
+   * @param {React.UIEvent} event The DOM event that triggered the change.
+   */
+  onToggleOpening: (event: React.UIEvent) => void;
```

Comment:
WDYT about naming the method (renaming `onOpen` and `onClose` accordingly) a bit differently to have a usual `onClick={handleClick}` scenario? 🤔 
- `handleOpenToggle`
- `handleToggleOpen`
- `toggleOpenState`
- `togglePickerOpen`

Reply:
`onOpen` and `onClose` are present in v7 so it's a small BC.
But if you think it's cleaner I'm fine doing it.

I'm not a fan of the `handleXXX` when they are provided by a hook (or as a prop).
I use them when creating variables like:


```tsx
const handleTogglePicker = () => { ... }

return <button onClick={handleTogglePicker}>Open me</button>
```

But when it comes from a button it feels off to me:

```tsx
const { handleTogglePicker } = usePickerFakeContext();

return <button onClick={handleTogglePicker}>Open me</button>
```

Not sure why :laughing: 
Maybe because `handleXXX` is a nomenclature used only for event handler and nothing prevents the user to use the methods from the hook in something else (in an effect, when a server request loads etc...).

For this kind of hook, if we don't want to `onXXX`, I would take to prefer a naming like `openPicker`, `closePicker` and `togglePickerOpening` for example.

Reply:
Great insights, especially regarding the possibility of this function being called in other places than the even handler. 👌 


> `onOpen` and `onClose` are present in v7 so it's a small BC.

They are and didn't even cross my eyes until you started working on this change, then it became a bit more clear, that we can improve clarity here. 🙈 
But they are present in the `valueResponse`.
Couldn't we update the naming only in the context? 🤔 

I would not mind having `openPicker`, `closePicker`, and `togglePickerOpen`(ing) (the `opening` word here sounds slightly strange 🙈 🤷 ).
P.S. Have we explored going for `setPickerOpen` instead of `onOpen` and `onClose`?
Noticed such an approch here:
https://github.com/adobe/react-spectrum/blob/e94e36431149b6d3e9ed5900de30b538c0110584/packages/%40react-stately/datepicker/src/useDatePickerState.ts#L55-L56

Reply:
> P.S. Have we explored going for setPickerOpen instead of onOpen and onClose?

That's yet another nomenclature indeed.

Reply:
WDYT about the following:
- `setPickerOpen`
- `openPicker`
- `closePicker`

We keep the 3 methods for maximal flexibility, we don't have any "handle" or "on" prefix.

Reply:
> But the context was introduced in a later v7 release I think 😬
So renaming them is a BC

Oh shoot, I forgot that we introduced it in v7 as well. 🙈 🥷 
Yeah, I'm all for refining the naming and not documenting too much on v7.
Maybe we can also skimp on having a codemod for the renaming. 😆 


> * `setOpen`
> * `openPicker`
> * `closePicker`

Didn't we plan to add a `togglePickerOpen` function to ease the simple use-case examples?
I had in mind if we considered having `setOpen` instead of `openPicker` and `closePicker` combination. 🤔

Reply:
Oh right, I got mixed up with my needs on the composition API here...

So if we start from the beginning, to modify the opening state of the picker, there are 4 potential methods:

1. Open
2. Closes
3. Toggles
4. Sets the provided value

Right now the context has 1. and 2.
I'm proposing to add 3. because it makes all the demos of custom fields easier to read.
And we will need 4. in the composition API because most modal / popover component take a prop like `onOpenChange` (Base UI and React Aria both do).

> I had in mind if we considered having setOpen instead of openPicker and closePicker combination. 🤔

So if I understand your comment correctly, you are proposing having 3 and 4 and dropping 1 and 2.

Reply:
Yes, your great summary is exactly what I had in mind. 👍 
WDYT about it? 🤔 
Is it worth the slight BC?
Does it make sense from the PoV of demos?

Reply:
I went for  a single `setOpen` method that matches the React `setState` methods.
If you prefer to have a dedicated `toggleOpen` I can of course add it, I feel like it's a nice balance, it keeps the public API clean and the demos concise:

```tsx
<IconButton onClick={() => pickerContext.setOpen((prev) => !prev)}>
  <DateRangeIcon />
</IconButton>
```